sdk/core/Azure.Core/src/Serialization/AzureObjectSerializer.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// An abstraction from reading typed objects.
+    /// </summary>
+    public abstract class AzureObjectSerializer
+    {
+        /// <summary>
+        /// Convert the provided value to it's binary representation and write it to <see cref="System.IO.Stream"/>.
+        /// </summary>
+        /// <param name="stream">The <see cref="System.IO.Stream"/> to write to.</param>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="inputType">The type of the <paramref name="value"/> to convert.</param>
+        public abstract void Serialize(Stream stream, object? value, Type inputType);
```

Comment:
On type part: One more alternative is make it generic method. 
Won't generic looks modern?

Reply:
See my response above.